src/bitExpert/PHPStan/Magento/Rules/AbstractModelRetrieveCollectionViaFactoryRule.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the phpstan-magento package.
+ *
+ * (c) bitExpert AG
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace bitExpert\PHPStan\Magento\Rules;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\ObjectType;
+
+/**
+ * Since 101.0.0 because collections should be used directly via factory
+ */
+class AbstractModelRetrieveCollectionViaFactoryRule implements Rule
+{
+    /**
+     * @return string
+     */
+    public function getNodeType(): string
+    {
+        return MethodCall::class;
+    }
+
+    /**
+     * @param Node $node
+     * @param Scope $scope
+     * @return array
+     * @throws ShouldNotHappenException
+     */
+    public function processNode(Node $node, Scope $scope): array
+    {
+        if (!$node instanceof MethodCall) {
+            throw new ShouldNotHappenException();
+        }
+
+        if (!$node->name instanceof Node\Identifier) {
+            return [];
+        }
+
+        if ($node->name->name !== 'getCollection') {
+            return [];
+        }
+
+        $type = $scope->getType($node->var);
+        $isAbstractModelType = (new ObjectType('Magento\Framework\Model\AbstractModel'))->isSuperTypeOf($type);
+        if (!$isAbstractModelType) {
```

Comment:
This won't work because it's an object, you need to ask `->no()`/`->maybe()`/`->yes()`.

Reply:
Make sure to test these rules with PHPStan itself :) This should get marked on level 4 I think.

Reply:
Thx. Weird, I ran this against a sample project and it seemed to have worked. Will check again.